### PR TITLE
[3006.x] deltaproxy, scheduler and event bus fixes

### LINKED ISSUE #64102: [BUG] One network device failure caused the restart proxy in loop until manually stopped.
**Description**
One network device failure caused the restart proxy in loop until manually stopped. 

**Setup**
Delta proxy devices loading from MongoDB script. There are 35 devices in one delta proxy and it keep crashing and unable to start.

Please be as specific as possible and give set-up details.
- [ ] VM (Virtualbox, KVM, etc. please specify)

**Steps to Reproduce the behavior**
```
/opt/saltstack/salt/lib/python3.10/site-packages/salt/transport/client.py:79: DeprecationWarning: This module is deprecated. Please use salt.channel.client instead.
  warn_until(
[CRITICAL] Unexpected error while connecting to salt-prod-master-1.infra.vmware.com
Traceback (most recent call last):
  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/minion.py", line 1135, in _connect_minion
    yield minion.connect_master(failed=failed)
  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/ext/tornado/gen.py", line 1056, in run
    value = future.result()
  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/ext/tornado/concurrent.py", line 249, in result
    raise_exc_info(self._exc_info)
  File "<string>", line 4, in raise_exc_info
  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/ext/tornado/gen.py", line 1064, in run
    yielded = self.gen.throw(*exc_info)
  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/minion.py", line 1376, in connect_master
    yield self._post_master_init(master)
  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/ext/tornado/gen.py", line 1056, in run
    value = future.result()
  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/ext/tornado/concurrent.py", line 249, in result
    raise_exc_info(self._exc_info)
  File "<string>", line 4, in raise_exc_info
  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/ext/tornado/gen.py", line 1070, in run
    yielded = self.gen.send(value)
  File "/opt/saltstack/salt/lib/python3.10/types.py", line 220, in send
    return self.__wrapped.send(val)
  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/metaproxy/deltaproxy.py", line 356, in post_master_init
    sub_proxy_data = subproxy_post_master_init(
  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/metaproxy/deltaproxy.py", line 524, in subproxy_post_master_init
    _proxy_minion.schedule.add_job(
  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/utils/schedule.py", line 413, in add_job
    evt.fire_event(
  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/utils/event.py", line 835, in fire_event
    self.pusher.send(msg)
  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/utils/asynchronous.py", line 125, in wrap
    raise exc_info[1].with_traceback(exc_info[2])
  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/utils/asynchronous.py", line 131, in _target
    result = io_loop.run_sync(lambda: getattr(self.obj, key)(*args, **kwargs))
  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/ext/tornado/ioloop.py", line 459, in run_sync
    return future_cell[0].result()
  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/ext/tornado/concurrent.py", line 249, in result
    raise_exc_info(self._exc_info)
  File "<string>", line 4, in raise_exc_info
  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/ext/tornado/gen.py", line 1070, in run
    yielded = self.gen.send(value)
  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/transport/ipc.py", line 456, in send
    yield self.stream.write(pack)
  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/ext/tornado/iostream.py", line 388, in write
    self._check_closed()
  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/ext/tornado/iostream.py", line 926, in _check_closed
    raise StreamClosedError(real_error=self.error)
salt.ext.tornado.iostream.StreamClosedError: Stream is closed
/opt/saltstack/salt/lib/python3.10/site-packages/salt/transport/client.py:79: DeprecationWarning: This module is deprecated. Please use salt.channel.client instead.
  warn_until(
```

**Expected behavior**
Proxy should start gracefully

**Versions Report**
Salt Version:
Salt: 3005

### LINKED ISSUE #64103: [BUG] Add multiple Network Devices without restarting proxy each time per device.
**Description**
Add multiple Network Devices without restarting proxy each time per device.  

**Setup**
Delta proxy devices loading from MongoDB script. There are 35 devices in one delta proxy and it requires 2-3 mins wait to accept key and it can be proceed only one device at a time.

Please be as specific as possible and give set-up details.
- [ ] VM (Virtualbox, KVM, etc. please specify)

**Steps to Reproduce the behavior**

**Expected behavior**
All devices keys should be accepted in one shot rather than serial acceptance and wait time.

**Screenshots**

**Versions Report**
Salt Version:
Salt: 3005

**Additional context**

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Update all the scheduler functions to include a fire_event argument which will determine whether to fire the completion event onto the event bus.  This event is only used when these functions are called via the schedule execution modules.  Inside deltaproxy, then update all scheudle related functions to include fire_event=False, as the event bus is not available when these functions are called.

### What issues does this PR fix or reference?
Fixes:

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/metaproxy/deltaproxy.py [hunk 1]
```diff
@@ -321,23 +334,34 @@ def post_master_init(self, master):
...
+        for future in concurrent.futures.as_completed(futures):
+            try:
+                sub_proxy_data = future.result()
+            except Exception as exc:  # pylint: disable=broad-except
+                _id = futures[future]
+                log.info(
+                    "An exception %s occured during initialization for %s, skipping.",
+                    exc,
+                    _id,
+                )
```
Comment (s0undt3ch) [inline salt/metaproxy/deltaproxy.py:362]:
```suggestion
                log.info(
                    "An exception occurred during initialization for %s, skipping: %s",
                    _id,
                    exc,
                )
```
The `str()` of an exception is sometimes verbose. It's probably more readable this way.

File: salt/metaproxy/deltaproxy.py [hunk 2]
```diff
@@ -347,16 +371,24 @@ def post_master_init(self, master):
...
+            try:
+                sub_proxy_data = subproxy_post_master_init(
+                    _id, uid, self.opts, self.proxy, self.utils
+                )
+            except Exception as exc:  # pylint: disable=broad-except
+                log.info(
+                    "An exception %s occured during initialization for %s, skipping.",
+                    exc,
+                    _id,
+                )
```
Comment (s0undt3ch) [inline salt/metaproxy/deltaproxy.py:389]:
```suggestion
                log.info(
                    "An exception occurred during initialization for %s, skipping: %s",
                    _id,
                    exc,
                )
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: tests/pytests/integration/proxy/conftest.py [hunk 3]
```diff
@@ -25,71 +25,53 @@ def deltaproxy_pillar_tree(request, salt_master, salt_delta_proxy_factory):
...
+            minion_id=minion_id,
+        )
 
+        controlproxy_pillar_file += """
+            - {}
+        """.format(
+            minion_id,
+        )
+
+    tempfiles = []
```
Comment (s0undt3ch) [inline tests/pytests/integration/proxy/conftest.py:63]: Why do we need this `tempfiles` variable?

File: tests/pytests/integration/proxy/conftest.py [hunk 4]
```diff
@@ -25,71 +25,53 @@ def deltaproxy_pillar_tree(request, salt_master, salt_delta_proxy_factory):
...
-    )
-    dummy_proxy_two_tempfile = salt_master.pillar_tree.base.temp_file(
-        "{}.sls".format(proxy_two), dummy_proxy_pillar_file
-    )
-    dummy_proxy_three_tempfile = salt_master.pillar_tree.base.temp_file(
-        "{}.sls".format(proxy_three), dummy_proxy_pillar_file
-    )
-    dummy_proxy_four_tempfile = salt_master.pillar_tree.base.temp_file(
-        "{}.sls".format(proxy_four), dummy_proxy_pillar_file
+    tempfiles = [top_tempfile, controlproxy_tempfile]
```
Comment (s0undt3ch) [inline tests/pytests/integration/proxy/conftest.py:68]: And this one?

File: changelog/64102.fixed.md [hunk 5]
```diff
@@ -0,0 +1,4 @@
+Update all the scheduler functions to include a fire_event argument which will determine whether to fire the completion event onto the event bus.
+This event is only used when these functions are called via the schedule execution modules.
+Inside deltaproxy, then update all scheudle related functions to include fire_event=False,
```
[the remark this answers is not in the document]
Comment (garethgreenaway) [reply changelog/64102.fixed.md:3]: No.  That does look incomplete.  Thanks for catching that.

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**